include/AutomationNode.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * AutomationNode.h - Declaration of class AutomationNode, which contains
+ *                       all information about an automation node
+ *
+ * Copyright (c) 2020 Ian Caio <iancaio_dev/at/hotmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef AUTOMATION_NODE_H
+#define AUTOMATION_NODE_H
+
+
+class AutomationPattern;
+
+
+class AutomationNode
+{
+public:
+	AutomationNode(); // Dummy constructor for the QMap
+	AutomationNode(AutomationPattern* pat, float value, int pos);
+	AutomationNode(AutomationPattern* pat, float inValue, float outValue, int pos);
+
+	AutomationNode& operator+=(float f)
+	{
+		m_inValue += f;
+		m_outValue += f;
+		return *this;
+	}
+	AutomationNode& operator-=(float f)
+	{
+		m_inValue -= f;
+		m_outValue -= f;
+		return *this;
+	}
+	AutomationNode& operator*=(float f)
+	{
+		m_inValue *= f;
+		m_outValue *= f;
+		return *this;
+	}
+	AutomationNode& operator/=(float f)
+	{
+		m_inValue/=f;
+		m_outValue/=f;
+		return *this;
+	}
+
+	inline float getInValue()
+	{
+		return m_inValue;
+	}
+	inline const float getInValue() const
+	{
+		return m_inValue;
+	}
```

Comment:
`inline float getInValue() const` is sufficient because modifying the return value doesn't change `this`. This also applies for other functions which returns `float`.